src/arrayvec.rs

```diff
@@ -500,11 +496,14 @@ impl<A: Array> ArrayVec<A> {
   /// ```
   #[inline(always)]
   pub fn try_push(&mut self, val: A::Item) -> Option<A::Item> {
-    if self.len == A::CAPACITY {
-      return Some(val);
-    }
+    debug_assert!(self.len <= A::CAPACITY);
```

Comment:
I'm not clear on what this debug_assert is for. If it's for nudging the codegen below it should be a full assert.

Reply:
Previously the check was `self.len == A::CAPACITY`, and if `self.len` was greater than `CAPACITY` it would panic on `self.data.as_slice_mut()[self.len]` few lines below.
The assert here is just to make sure that we don't set somewhere `self.len` accidentally above `CAPACITY`.
It is debug only, to cut the possible panic branch on release, since if we should keep the vec always internally consistent.

Reply:
In benchmarks it doesn't do much, but still, the function is smaller, so theoretically it inlines better

Reply:
okay, right, as long as it's not actually important it can be a debug assert.

without the assert we'd just get a panic elsewhere i guess.